### DISCUSSION TITLE
How should be used 'Compare' to get result without 'shadow' of the second image?

### DISCUSSION THREAD

**Discussion by fenoktech** [dlemstra/Magick.NET#1020] [Help]
Hey there,

How can we implement the following function using Magick.Net; as provided by command line version of ImageMagick?

[Compare: ](https://legacy.imagemagick.org/Usage/compare/#compare) https://legacy.imagemagick.org/Usage/compare/#compare

> If you don't want that 'shadow' of the second image, from IM v6.4.2-8 you can add a "-compose src" to the options to remove it. 
> compare bag_frame1.gif bag_frame2.gif \
>            -compose Src compare_src.gif

 

I couldn’t find a way to use a such parameter/option like “-compose Src” to remove the 'shadow' of the second image in the result of the “diffImage”.

```
MagickImage image1 = new MagickImage(fileName1)
MagickImage image2 = new MagickImage(fileName2)
MagickImage diffImage = new MagickImage();

CompareSettings compareSettings = new CompareSettings
                    {
                        HighlightColor = MagickColor.FromRgb(255, 255, 0),
                        LowlightColor = MagickColor.FromRgb(0,255,0),
                        Metric = ErrorMetric.Fuzz
                    };

double distortion = image1.Compare(image2, compareSettings, diffImage, Channels.All);

using (MemoryStream memoryStream = new MemoryStream())
                        {
                            diffImage.Write(memoryStream, MagickFormat.Png);
                            pictureBox.Image = Image.FromStream(memoryStream);
                        }

                        Debug.WriteLine(distortion);
```

Any help is appreciated.
Thanks.

**Comment by dlemstra** [comment]
What happens when you set `diffImage.Compose` to `CompositeOperator.Src`?

**Reply by fenoktech** [reply]
I’ve tried setting `diffImage.Compose` to `CompositeOperator.Src` and nothing changed unfortunately..
Then I have added some other properties as well to see what happens with them. Again, nothing changed in the result of “diffImage”.
(Not sure I have used them correctly by the way.)

```
MagickImage image1 = new MagickImage(fileName1)
MagickImage image2 = new MagickImage(fileName2)

//MagickImage diffImage = new MagickImage();
//diffImage.Compose = CompositeOperator.Src;

MagickImage diffImage = new MagickImage
{
     Compose = CompositeOperator.Src,
     BorderColor = MagickColor.FromRgb(255, 0, 0),
     Label = "Warning"
};

CompareSettings compareSettings = new CompareSettings
{
     HighlightColor = MagickColor.FromRgb(255, 255, 0),
     LowlightColor = MagickColor.FromRgb(0,255,0),
     Metric = ErrorMetric.Fuzz
};

double distortion = image1.Compare(image2, compareSettings, diffImage, Channels.All);

using (MemoryStream memoryStream = new MemoryStream())
{
     diffImage.Write(memoryStream, MagickFormat.Png);
     pictureBox.Image = Image.FromStream(memoryStream);
}

Debug.WriteLine(distortion);

```

Thanks,

**Reply by dlemstra** [reply]
You will need to set the `Compose` on the `image1` instead.

**Reply by fenoktech** [reply]
Yes! There is no shadow no more..
You're great, thank you very much indeed. :)
Well, instead of 'shadow', I get the 'differences' on the RGB(211,211,211) background.
Is it supposed to be so? Or, is there a way to change this (let's say) 'predefined background color' as well?
Thanks again.

**Reply by dlemstra** [reply]
I am sorry but I don't understand what you are tring to ask me.
